memgpt/agent.py

```diff
@@ -184,21 +201,32 @@ def __init__(
         if "system" not in agent_state.state:
             raise ValueError(f"'system' not found in provided AgentState")
         self.system = agent_state.state["system"]
-
+        if "system_template" not in agent_state.state:
+            self.system_template = ""
+        else:
+            self.system_template = agent_state.state["system_template"]
+        if "system_template_fields" not in agent_state.state:
+            self.system_template_fields = {}
+        else:
+            self.system_template_fields = agent_state.state["system_template_fields"]
         if "functions" not in agent_state.state:
             raise ValueError(f"'functions' not found in provided AgentState")
         # Store the functions schemas (this is passed as an argument to ChatCompletion)
         self.functions = agent_state.state["functions"]  # these are the schema
         # Link the actual python functions corresponding to the schemas
         self.functions_python = {k: v["python_function"] for k, v in link_functions(function_schemas=self.functions).items()}
         assert all([callable(f) for k, f in self.functions_python.items()]), self.functions_python
-
-        # Initialize the memory object
-        if "persona" not in agent_state.state:
-            raise ValueError(f"'persona' not found in provided AgentState")
-        if "human" not in agent_state.state:
-            raise ValueError(f"'human' not found in provided AgentState")
-        self.memory = initialize_memory(ai_notes=agent_state.state["persona"], human_notes=agent_state.state["human"])
+        if "core_memory_type" in agent_state.state and agent_state.state["core_memory_type"] == "custom":

```

Comment:
If we make changes to make the entire system block templatable, maybe it makes more sense to call this field something like `system_template_type`? Or have both?

Reply:
I think keeping both would make sense, core_memory_type for the type of core memory and system_message_layout_template
core_memory_section_template
For templating the layout of the system message with memory overview. And core_memory_section_template for templating each section, like persona or human.